src/basic/stat-util.c

```diff
@@ -205,7 +205,9 @@ int path_check_fstype(const char *path, statfs_f_type_t magic_value) {
 
 bool is_temporary_fs(const struct statfs *s) {
     return is_fs_type(s, TMPFS_MAGIC) ||
-           is_fs_type(s, RAMFS_MAGIC);
+           is_fs_type(s, RAMFS_MAGIC) ||
+           is_fs_type(s, CGROUP_SUPER_MAGIC) ||
+           is_fs_type(s, CGROUP2_SUPER_MAGIC);
```

Comment:
`rm_rf(path, REMOVE_ROOT|REMOVE_ONLY_DIRECTORIES)` doesn't work if `path` on the `cgroup[2]`-fs. I think we should fix the `rm_rf` instead. But let's check that this PR really fixes #4181 first.